### PR TITLE
Fix potential null pointer deferenced

### PR DESCRIPTION
**Pull request by raniervf**
clist_count is a MACRO:
#define     clist_count(lst)               ((lst)->count)

*fetch_result is check agaist NULL right below.

### CODE REVIEW

File: src/low-level/imap/qresync.c [hunk 1]
```diff
@@ -409,7 +409,7 @@ int mailimap_fetch_qresync_vanished(mailimap * session,
     * p_vanished = get_vanished(session);
   }
 
-  if (clist_count(* fetch_result) == 0) {
+  if (* fetch_result != NULL && clist_count(* fetch_result) == 0) {
```
Comment (dinhviethoa) [inline src/low-level/imap/qresync.c:412]: if `(* fetch_result == NULL)`, I think we should also do the same thing here.

Could you change the test to the following instead?

```
if (* fetch_result == NULL || clist_count(* fetch_result) == 0) {
  // make the compiler happy by checking value of * fetch_result but it's unlikely to happen.
  error_code = response->rsp_resp_done->rsp_data.rsp_tagged->rsp_cond_state->rsp_type;
} else {
   error_code = MAILIMAP_RESP_COND_STATE_OK;
}
```

File: src/low-level/imap/qresync.c [hunk 2]
```diff
@@ -487,7 +487,7 @@ int mailimap_uid_fetch_qresync_vanished(mailimap * session,
...
```
Comment (dinhviethoa) [inline src/low-level/imap/qresync.c:490]: Probably similar here.

File: src/low-level/imap/qresync.c (hunk 1 above)
Comment (raniervf) [reply src/low-level/imap/qresync.c]: You right, here.

File: src/low-level/imap/qresync.c (hunk 2 above)
Comment (raniervf) [reply src/low-level/imap/qresync.c]: You right here.

**Pull request merged**